src/input/accessors.hh

```diff
@@ -877,12 +844,22 @@ struct TypeDispatch<FullEnum> {
     static inline ReadType value(const Address &a, const InputType &t) { return ReadType( a.storage_head()->get_int(), t ); }
 };
 
-template<>
-struct TypeDispatch<int> {
+template<class T>
+struct TypeDispatch<T, typename boost::enable_if<boost::is_integral<T> >::type> {
     typedef Input::Type::Integer InputType;
     typedef const int ReadType;
     typedef int TmpType;
-    static inline ReadType value(const Address &a, const InputType&) { return a.storage_head()->get_int(); }
+    static inline ReadType value(const Address &a, const InputType&t) {
+    	std::int64_t val = a.storage_head()->get_int();
+    	if (val >= -(std::int64_t)std::numeric_limits<unsigned int>::max() &&
```

Comment:
No,no. The value is already checked against  InputType bounds in JSONToStorage, we have to check 
that range of target type T is larger or equal to the range on InputType.

Also add explicit conversion to the target type T.